### ISSUE TITLE
"warning: message: high error 2311: glGetBooleanv parameter has an invalid value '8': must be less than 8 (GL_INVALID_VALUE)" on call to glDrawElements

### ISSUE THREAD

**Issue by AnyOldName3** [apitrace/apitrace#592]
I'm trying to fix a call to `glDrawElements` which is causing a `GL_INVALID_OPERATION` error, but QApiTrace is filling the call's error list with `warning: message: high error 2311: glGetBooleanv parameter <active texture> has an invalid value '8': must be less than 8 (GL_INVALID_VALUE)` with values from 8 to 15, which is distracting. Obviously, `glDrawElements` is not `glGetBooleanv`, so cannot be the source of the extra warnings.

My guess is that when QApiTrace is working out which texture units have active textures to display the OpenGL state for the call, it's not asking the driver which numbers can be added to `GL_TEXTURE0` without causing problems (or the driver is giving a value usable in other situations, but not how it's being used here), and that the warnings are erroneously being bundled with those caused by the call in the traced application. I reckon it might be https://github.com/apitrace/apitrace/blob/9b9eb3d955b2464006a0790470be6d8f8ee14fe6/retrace/glstate_images.cpp#L950 - if texture units above 8 are being bound and having their parameters queried, but my driver only likes telling you which texture type is bound for old OpenGL versions when the currently bound texture unit is 7 or lower, I guess it could be doing what I'm seeing.

Basically, I think the cause of the extra warnings is just my GPU driver being stupid rather than QApiTrace doing something wrong, but whatever the reason, QApiTrace should only show errors caused by calls in the application being traced rather than all calls including its own.

**Comment by AnyOldName3** [comment]
After reading something else today while doing something else, I've realised this might be because the loop iterates from `0` to `GL_MAX_COMBINED_TEXTURE_IMAGE_UNITS`, but texture unit state (rather than texture image unit state) is being queried. It might be better to stop the iteration at `GL_MAX_TEXTURE_UNITS` instead (and potentially then do a different iteration from `GL_MAX_TEXTURE_UNITS` to `GL_MAX_COMBINED_TEXTURE_IMAGE_UNITS` querying only texture image unit state).

There's still the issue of QApiTrace displaying OpenGL errors from QApiTrace calls as if they were caused by the traced application. That's probably best solved by doing one batch of error collection right after the application call, and then another after recording the OpenGL state, with only the first batch being shown to the user.
